app/models/miq_region.rb

```diff
@@ -154,6 +158,8 @@ def self.replication_type
   end
 
   def self.replication_type=(desired_type)
+    return unless PglogicalSubscription.logical_replication_supported?
+
```

Comment:
It really feels like this entire method doesn't belong here at all.  MiqRegion shouldn't have this orchestration code in it and should instead defer to some method in MiqPglogical.. I'm thinking something like...

```ruby
# Here

def self.replication_type=(desired_type)
    current_type = replication_type
    return desired_type if desired_type == current_type

    MiqPglogical.new.replication_type = desired_type
end
```

Then once that's in place, `MiqPglogical.new` itself can use the null object pattern as described by @kbrock

Reply:
would prefer a factory implement null object pattern rather than `new` -- that is such a ruby hack. gives me the hee-bee-jee-bees

Reply:
Yeah, I like this.  It's nice that guard clauses make this more obvious where it's in the wrong context.  I didn't see this at first but totally agree after being away from the code for a bit.

Reply:
Ok, moved.